File: set_1/easy/python/d884d311-a562-4321-8e4f-1bf6bde749c5.py
# Solution 1:

def max_min_rearranger(arr):

    arr_len = len(arr)
    temp_arr = []

    min_index, max_index = 0, arr_len-1

    flag = True

    for i in range(arr_len):
        if flag is True:
            temp_arr[i] = arr[max_index]
            max_index -= 1
        else:
            temp_arr[i] = arr[min_index]
            min_index += 1

        flag = bool(1-flag)

        return temp_arr


arr1 = [24, 25, 29, 40, 43, 46, 56, 66, 82, 95]

print(f"Original input array --> {arr1}")
res1 = max_min_rearranger(arr1)
print(f"The max-min sorted array form --> {res1}")

File: set_1/arrays/javascript/d884d311-a562-4321-8e4f-1bf6bde749c5.js
// Solution 1:

const maxMinRearrager = arr => {
    const arrLen = arr.length;
    let tempArr = [];
    let [minIndex, maxIndex] = [0, arrLen - 1];

    let flag = true;
    for (let i = 0; i < arrLen; i++) {
        if (flag) {
            tempArr[i] = arr[maxIndex--];
        }
        else {
            tempArr[i] = arr[minIndex++];
        }
        flag = !flag;
    }
    return tempArr;
}

const arr1 = [24, 25, 29, 40, 43, 46, 56, 66, 82, 95];
const res1 = maxMinRearrager(arr1);
console.log(`Original input array --> [${arr1}]`)
console.log(`The max-min sorted array form --> [${res1}]`);



// Solution 2:

const maxMinRearrager2 = arr => {

    let arrLength = arr.length - 1;

    for (let i = 0; i <= arrLength - 1; i++) {
        if (i % 2 == 0 && i <= arrLength) {
            arr.splice(i, 0, arr[arrLength]);
            i++;
        }
    }
    return (arr.splice(0, arrLength + 1))
}

const arr2 = [24, 25, 29, 40, 43, 46, 56, 66, 82, 95];
console.log(`Original input array --> [${arr2}]`)
const res2 = maxMinRearrager2(arr2);
console.log(`The max-min sorted array form --> [${res2}]`);